cpp/mrc/src/internal/data_plane/data_plane_resources.cpp

```diff
@@ -415,11 +421,20 @@ void DataPlaneResources2::decrement_tokens(remote_descriptor::RemoteDescriptorDe
         remote_descriptor->encoded_object().set_tokens(tokens);
         if (tokens == 0)
         {
-            m_remote_descriptor_by_id.erase(dec_message->object_id);
+            {
+                std::unique_lock lock(m_remote_descriptors_mutex);
+                m_remote_descriptor_by_id.erase(dec_message->object_id);
+            }
+            m_remote_descriptors_cv.notify_one();
         }
     }
 }
 
+void DataPlaneResources2::set_max_remote_descriptors(uint64_t max_remote_descriptors)
+{
+    m_max_remote_descriptors = max_remote_descriptors;
```

Comment:
If this can change while the tool is running, we probably need a `m_remote_descriptors_cv.notify_all();` to check any waiters.

Reply:
Makes sense, done in https://github.com/nv-morpheus/MRC/pull/447/commits/ba9be708fe523917954d0302155e95672c14c753, also add a test parameter to test for this condition as well in https://github.com/nv-morpheus/MRC/pull/447/commits/33aa01ca9bbe9c11b91bbe3b2e9db7771f596b01 .